.github/workflows/build.yml

```diff
@@ -31,11 +31,13 @@ jobs:
       - name: Install Python
         uses: actions/setup-python@v5
         with:
-          python-version: "3.11"
+          python-version: "3.12"
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          pip install ruff
+          python -m pip install uv
+          uv venv
+          source .venv/bin/activate
+          uv sync
       # Update output format to enable automatic inline annotations.
       - name: Run Ruff
         run: |
```

Comment:
Below, I'd suggest `uv run ruff ...` to ensure that the ruff installed from within the uv virtualenv is being used.

Reply:
And e.g. also for the Python license file, use `uv run python ./dev/create_license.py` if that script requires any uv-provided dependencies.